common/junit-platform-native/src/main/java/org/graalvm/junit/platform/config/jupiter/JupiterConfigProvider.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.jupiter.params.converter.ConvertWith;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.graalvm.junit.platform.JUnitPlatformFeature.debug;
+
+public class JupiterConfigProvider implements PluginConfigProvider {
+
+    @Override
+    public void onLoad(NativeImageConfiguration registry) {
+        String[] buildTimeInitializedClasses = new String[]{
+                "org.junit.jupiter.engine.config.EnumConfigurationParameterConverter",
+                "org.junit.jupiter.engine.descriptor.ClassTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.TestFactoryTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.JupiterTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.JupiterTestDescriptor$1",
+                "org.junit.jupiter.engine.descriptor.MethodBasedTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor",
+                "org.junit.jupiter.engine.descriptor.TestTemplateTestDescriptor",
+                "org.junit.jupiter.engine.execution.ConditionEvaluator",
+                "org.junit.jupiter.engine.execution.ExecutableInvoker"
+        };
+        for (String className : buildTimeInitializedClasses) {
+            registry.initializeAtBuildTime(className);
+        }
+    }
+
+    @Override
+    public void onTestClassRegistered(Class<?> testClass, NativeImageConfiguration registry) {
+        AnnotationUtils.registerClassesFromAnnotationForReflection(testClass, registry, ArgumentsSource.class, source -> new Class[]{source.value()});
+        AnnotationUtils.registerClassesFromAnnotationForReflection(testClass, registry, ExtendWith.class, ExtendWith::value);
+        AnnotationUtils.forEachAnnotatedMethod(testClass, EnumSource.class, (m, annotation) -> handleEnumSource(m, annotation, registry));
+        handleTestOrderer(testClass, registry);
+        AnnotationUtils.registerClassesFromAnnotationForReflection(testClass, registry, DisplayNameGeneration.class, source -> new Class[]{source.value()});
+        AnnotationUtils.registerClassesFromAnnotationForReflection(testClass, registry, IndicativeSentencesGeneration.class, source -> new Class[]{source.generator()});
+        AnnotationUtils.forEachAnnotatedMethodParameter(testClass, ConvertWith.class, annotation -> registry.registerAllClassMembersForReflection(annotation.value()));
+        AnnotationUtils.forEachAnnotatedMethodParameter(testClass, AggregateWith.class, annotation -> registry.registerAllClassMembersForReflection(annotation.value()));
```

Comment:
For things like `@ConvertWith` and `@AggregateWith`, why do we need to register all class members for reflection?

Don't we just need to register the class and its default constructor for reflection?

Reply:
For `ConvertWith` and `AggregateWith`, we shouldn't need to register anything apart from the class and the default constructor. However, I've noticed that for some other annotations, we may need to register additional methods. For example, for any annotation consumer, we may also need:
```
org.junit.jupiter.params.support.AnnotationConsumerInitializer#isAnnotationConsumerAcceptMethod
```
To be on the safe side, I've went with registering everything. We can in the future do registration on a more fine-grained level